base/inference.jl

```diff
@@ -1400,10 +1398,6 @@ function abstract_call_gf_by_type(f::ANY, atype::ANY, sv::InferenceState)
         add_mt_backedge(ftname.mt, argtype, sv)
         update_valid_age!(min_valid[1], max_valid[1], sv)
     end
-    if isempty(applicable)
-        # TODO: this is needed because type intersection is wrong in some cases
-        return Any
```

Comment:
Oh, it's not type intersection this time (the failing arrayops test), see #21769. Indeed, there should be a `MethodError`, but there isn't.